lib/connectors/base/registry.rb

```diff
@@ -20,6 +20,18 @@ def register(name, klass)
       def connector_class(name)
         @connectors[name]
       end
+
+      def connector(name, params = nil)
+        connector_class = connector_class(name)
+        if connector_class.present?
+          return params.present? ? connector_class.new(params) : connector_class.new
+        end
+        raise "Connector #{name} is not yet registered. You need to register it before use"
```

Comment:
you could probably put this raise at the beginning of this function, ending the line with `unless connector_class.present?`